testbed/integration/config_examples_test.go

```diff
@@ -202,3 +203,113 @@ func TestConfigJaegerGrpc(t *testing.T) {
 	// assert
 	tc.ValidateData()
 }
+
+func TestConfigHistogramTransform(t *testing.T) {
+	// arrange
+	col := testbed.NewChildProcessCollector(testbed.WithAgentExePath(CollectorTestsExecPath))
+	cfg, err := os.ReadFile("../../config_examples/split_histogram.yaml")
+	require.NoError(t, err)
+
+	receiverPort := testbed.GetAvailablePort(t)
+	exporterPort := testbed.GetAvailablePort(t)
+
+	parsedConfig := string(cfg)
+	parsedConfig = replaceOtlpGrpcReceiverPort(parsedConfig, receiverPort)
+	parsedConfig = replaceDynatraceExporterEndpoint(parsedConfig, exporterPort)
+
+	configCleanup, err := col.PrepareConfig(parsedConfig)
+	require.NoError(t, err)
+	t.Cleanup(configCleanup)
+
+	actualMetricsData := pmetric.NewMetrics()
+	actualMetrics := actualMetricsData.ResourceMetrics().AppendEmpty().ScopeMetrics().AppendEmpty().Metrics()
+
+	expectedMetricData := pmetric.NewMetrics()
+	expectedMetrics := expectedMetricData.ResourceMetrics().AppendEmpty().ScopeMetrics().AppendEmpty().Metrics()
+
+	startTime := time.Now()
+	// Metric start time will be set to the end of the previous collection cycle for DELTA metrics
+	startTimeStamp := pcommon.NewTimestampFromTime(startTime)
+	// Add a default PeriodicExportingMetricReader interval (30s)
+	timeStamp := pcommon.NewTimestampFromTime(startTime.Add(time.Second * 30))
+
+	// Histogram
+	histogram := actualMetrics.AppendEmpty()
+	histogram.SetName("my.histogram")
+	histogram.SetUnit("custom_unit")
+	histogram.SetDescription("My custom histogram")
+	histogram.SetEmptyHistogram()
+	histogram.Histogram().SetAggregationTemporality(pmetric.AggregationTemporalityDelta)
+	histogramDataPoint := histogram.Histogram().DataPoints().AppendEmpty()
+	histogramDataPoint.Attributes().PutStr("key", "value")
+	histogramDataPoint.SetStartTimestamp(startTimeStamp)
+	histogramDataPoint.SetTimestamp(timeStamp)
+
+	// recorded 0.5, 3.5, 3.5
+	histogramDataPoint.ExplicitBounds().Append(1, 2, 3)
+	histogramDataPoint.SetCount(3)
+	histogramDataPoint.SetSum(7.5)
+	histogramDataPoint.SetMin(0.5)
+	histogramDataPoint.SetMax(3.5)
+	histogramDataPoint.BucketCounts().Append(1, 0, 0, 0, 2)
+
+	// Split Histogram
+	splitCountMetric := expectedMetrics.AppendEmpty()
+	splitCountMetric.SetName("my.histogram_count")
+	splitCountMetric.SetUnit("custom_unit")
```

Comment:
**Note for reviewers:** does it make sense to have the `unit` be the same for `count` metrics split from `histogram` metrics?

Take [`http.server.request.duration` metric](https://opentelemetry.io/docs/specs/semconv/http/http-metrics/#http-server) for example, it records the duration of requests on the server. The unit for that metric is `s` (seconds), however, the `count` property on that histogram is actually the amount of requests recorded, not seconds (that's the sum, which is the total time the server spent on handling requests of that route, for instance).

Reply:
I opened an issue upstream as I think it's applicable to all metrics that can be passed to that function: https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/31575

Reply:
I suggest we go ahead with adding the tests as-is then iterate on them based on the outcome of https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/31575 

WDYT @evan-bradley @arminru?